_docs/cardano/2017-01-01-proof-of-stake.md

```diff
@@ -9,83 +9,173 @@ visible: true
+```
+
+It means that the number of slot leaders is strictly equal to the number of slots
+in epoch (let's call this number `N`), so it is impossible to produce more than `N`
+blocks during an epoch.
+
+If slot leader missed his slot (for example, he was offline at that moment), he
+lost his right to produce a block until he will be elected again. So one or more
+slots can remain empty (i.e. without generated blocks), but please note that the
+most of blocks (i.e. at least 50% + 1) **must** be generated during an epoch.
+
+## Slot Leaders Election
+
+Slot leaders are elected from among stakeholders. Please note that not all stakeholders
+participate in this election, but only ones who has enough stake (for example, 2% of
+the total stake). Let's call this group of stakeholders "electors".
+
+During an epoch electors elect slot leaders for the next epoch. Thus, at the end of epoch
+`N` it is already known who are slot leaders for the epoch `N+1`, and it cannot be
+changed.
+
+You can think of this election as a "fair lottery": anyone from stakeholders can become
+a slot leader. But important idea of PoS is that the more stake stakeholder has, the more
+chances one has to be elected as a slot leader. 
+
+Please note that one stakeholder can be elected as a slot leader for more than one slot
+during the same epoch.
+
+### Multiparty Computation
+
+The fundamental problem of election process is its unbiasedness. We need some randomness
+as a base for election, in this case results of this election will be random and fair.
+So the question is where this randomness can be obtained from?
+
+We are using multiparty computation (MPC) approach: each elector independently performs an
+action which can be called "coin tossing" and after that shares result with other electors.
+The idea is that results are randomly generated by each elector, but eventually they agree on
+the same final value.
+
+#### Commitment Phase
+
+First of all, elector generates a secret (special random value). Then elector forms a
+"commitment" which is a message that contains encrypted shares and proof of secret.
+
+Then elector signs this commitment with its secret key, specifies epoch's number and attaches
+its public key. In this case everybody can check who created this commitment and which epoch
+this commitment is for.
+
+After that elector send its commitment to other electors, so eventually each elector collects
+commitments from all other electors.
+
+#### Opening Phase
+
+Now elector sends an "opening", special value for opening a commitment. You can think of a
+commitment as a locked box (with a secret in it), and opening is a key we need to open this
+box and get the secret from it.
+
+So eventually elector has commitments and openings. Then elector verifies that commitments
+and openings are match, and if so, he gets the secrets from commitment and forms a seed from
+these secrets. So all electors get the same seed, and it will be used for Follow the Satoshi
+algorithm.
```

Comment:
What about Shares Phase? Is it intentionally omitted for simplicity?

Reply:
Yes, for simplicity. Do you think I must add it?

Reply:
I think it'd be good to add it, at least one or two sentences, because otherwise someone who will also read Ouroboros paper might be confused. There is no need to describe details, just write that Shares Phase exists to prevent possible confusion.